### PR TITLE
Lazy load images

### PR DESCRIPTION
**Pull request by ilias-t**
[][demo] | Fixes RM-3151
:---:|:---:

## 🧰 Changes

- lazy load images
- default to "smart" property for width
- filter out `undefined` img props

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-411.herokuapp.com
[prod]: https://developers.readme.io

### CODE REVIEW

File: __tests__/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1018,15 +1018,15 @@ exports[`export multiple Markdown renderers renders plainText from AST 1`] = `"H
...
 
-exports[`image 1`] = `"<p><span aria-label=\\"Image\\" class=\\"img\\" role=\\"button\\" tabindex=\\"0\\"><img src=\\"http://example.com/image.png\\" alt=\\"Image\\" align=\\"\\" caption=\\"\\" height=\\"auto\\" title=\\"\\" width=\\"auto\\"><span class=\\"lightbox\\" role=\\"dialog\\" tabindex=\\"0\\"><span class=\\"lightbox-inner\\"><img src=\\"http://example.com/image.png\\" align=\\"\\" caption=\\"\\" height=\\"auto\\" title=\\"Click to close...\\" width=\\"auto\\" alt=\\"Image\\" class=\\"lightbox-img\\" loading=\\"lazy\\"></span></span></span></p>"`;
+exports[`image 1`] = `"<p><span aria-label=\\"Image\\" class=\\"img\\" role=\\"button\\" tabindex=\\"0\\"><img src=\\"http://example.com/image.png\\" alt=\\"Image\\" align=\\"\\" caption=\\"\\" height=\\"auto\\" title=\\"\\" width=\\"smart\\" loading=\\"lazy\\"><span class=\\"lightbox\\" role=\\"dialog\\" tabindex=\\"0\\"><span class=\\"lightbox-inner\\"><img src=\\"http://example.com/image.png\\" align=\\"\\" caption=\\"\\" height=\\"auto\\" title=\\"Click to close...\\" width=\\"smart\\" alt=\\"Image\\" class=\\"lightbox-img\\" loading=\\"lazy\\"></span></span></span></p>"`;
```
Comment (ilias-t) [inline __tests__/__snapshots__/index.test.js.snap:1021]: omg this diff is painful to look at with all this red 🟥

File: components/Image/index.jsx [hunk 2]
```diff
@@ -106,7 +108,7 @@ Image.defaultProps = {
   height: 'auto',
   src: '',
   title: '',
-  width: 'auto',
+  width: 'smart',
```
Comment (ilias-t) [inline components/Image/index.jsx:111]: thanks for the suggestion @kellyjosephprice This should map to [this line](https://github.com/readmeio/markdown/blob/78e5fd51202fb8b7a0cce527a84ef2313a7ccdf2/components/Image/style.scss#L43)

Comment (rafegoldberg) [inline components/Image/index.jsx:111]: Why this change? I'd prefer keeping `auto`, unless there's a reason to use something else:
```suggestion
  width: 'auto',
```

File: __tests__/__snapshots__/index.test.js.snap (hunk 1 above)
Comment (rafegoldberg) [reply __tests__/__snapshots__/index.test.js.snap:1021]: MY EYES

File: components/Image/index.jsx (hunk 2 above)
Comment (rafegoldberg) [reply components/Image/index.jsx:111]: Ahhh, [I see now](https://github.com/readmeio/markdown/pull/411#discussion_r779953681). Yeah, I don't think we actually want this to be the default; we *want* them to use the browser's intrinsic `auto` sizing!

Comment (ilias-t) [reply components/Image/index.jsx]: ok yeah, to your point that we don't want to change this and then have default styles mess up people's sites—going to revert this back to auto from smart

**Review by kellyjosephprice — CHANGES_REQUESTED** [review]
It looks like some refactoring I did was too clever:
https://github.com/readmeio/markdown/blob/next/processor/parse/magic-block-parser.js#L111

It looks like `imgWidthsBySize[img.sizing]` is converting a proper `undefined` to a string `'undefined'`. I guess it shouldn't be surprising that the property access has to convert `undefined` to a string?!?!

**Review by kellyjosephprice — APPROVED** [review]
Thanks!

**Review by kellyjosephprice — APPROVED** [review]

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.38.0-next.1</b>](/readmeio/markdown/releases/tag/v6.38.0-next.1)
<br>

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.38.0</b>](/readmeio/markdown/releases/tag/v6.38.0)
<br>
